downloader/downloader/main.py

```diff
@@ -687,3 +699,69 @@ def stop():
 		worker.get() # re-raise error if failed
 
 	logging.info("Gracefully stopped")
+
+class SegmentLinker(SegmentGetter):
```

Comment:
This is a style nitpick but I think the job of `SegmentGetter` and `SegmentLinker` is sufficiently different that `Linker` shouldn't inherit from `Getter`. Maybe it makes sense for them to have a common base class, or maybe everything HTTP-related should be moved out of `SegmentGetter` and then `SegmentGetter` is the common code and there's specific "local link" and "http fetch" providers (similar to how we have a twitch provider, youtube provider etc instead of a `TwitchStreamWorker` subclass of `StreamWorker`)

Reply:
I can do that. I mostly did it the way I did to reuse code with the smallest diff footprint, but I can definitely see that being better.